### DISCUSSION TITLE
How to extract digitally signed name from the PDF

### DISCUSSION THREAD

**Discussion by sunov** [pymupdf/PyMuPDF#1541] [General]
How to extract digitally signed name from the PDF using pymupdf

**Comment by JorjMcKie** [comment]
This is not supported in PyMuPDF yet. I also currently have no plan whether or when to implement it, sorry.

**Reply by sunov** [reply]
Thanks

**Reply by sunov** [reply]
@JorjMcKie  Currently, We have been using 1.16.7 pymupdf version for more than two years. Planning to use 1.19.4, Could you please mention what are the major issues fixed from 1.16.7 to 1.19.4. Observed following issues in 1.16.7

1.Digitized pdf returns only: 􀁌 as text.
2.Text is broken into single character[LUN, RR, I, CA....]
3.Bad order of text for Hebrew documents

**Reply by JorjMcKie** [reply]
I am maintaining a meticulous list of changes per version and also per single patch. Please find this information [here](https://pymupdf.readthedocs.io/en/latest/changes.html).

**Reply by sunov** [reply]
@JorjMcKie Thank you very much. I will go through the changes.

**Reply by sunov** [reply]
1.Digitized pdf returns only: 􀁌 as text.
2.Text is broken into single character[LUN, RR, I, CA....]
3.Bad order of text for Hebrew documents

@JorjMcKie  These issues fixed in 1.19.4 version

**Reply by JorjMcKie** [reply]
Even a lot earlier: did not touch any of this in 1.19.*, probably even not in 1.18.*.

**Reply by sunov** [reply]
@JorjMcKie , Able to extract digitally signed signature from pdf with 1.19.4 version. Not Sure, In which version the feature added. Thanks a lot.

**Reply by manipes-ag** [reply]
Hi @sunov, I'm also currently trying to extract digital signature and its metadata. I'm also looking for ways to extract this using PyMuPDF. Can you please share some code snippets on how you achieved it? Thanks
